### ISSUE TITLE
[Button]: Disabled ghost icon-only Button missing associated label

### ISSUE THREAD

**Issue by zJag** [carbon-design-system/carbon#5741]
## Environment

> Browser

Chrome (others are probably effected)

> Automated testing tool and ruleset

DAP Chrome Extension

## Detailed description

> What version of the Carbon Design System are you using?

v10.10.0

> What did you expect to happen?

Using an icon-only ghost Button from `carbon-components-react` should not cause violations with DAP. 

> What happened instead?

When disabled, the button causes a violation with DAP stating the associated label is missing.  

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Render an icon-only ghost button with proper props.
2. Disable the button.
3. Run the DAP extension

[CodeSandbox](https://codesandbox.io/s/red-forest-4yyku)

## Additional information

- The styles introduced with https://github.com/carbon-design-system/carbon/pull/5313 hides the assistive text tooltip when the button is a ghost button and is disabled. 
- The icon within the button receives an `aria-hidden` attribute.
- Using [W3C accessible name computation](https://www.w3.org/TR/accname-1.1/#mapping_additional_nd_te) the button's name would be blank.

This could be fixed by not hiding the assistive text when the button is disabled (in `carbon-components`). This could also be fixed in `carbon-components-react` by ensuring the icon-only button receives an accessible name. For the latter, I was able to create a fix on my forked branch, and would be able to submit a PR if that's the proper solution.

**Comment by tw15egan** [comment]
Since the rest of the buttons still have the assistive text when disabled, we should just remove the styles that are specific to the ghost button. PR Incoming 🚀 

https://github.com/carbon-design-system/carbon/pull/5313/files#diff-b73d9732aa1bfa4b943f188fdd4bfeccR228-R232
